pandas/_libs/window/aggregations.pyx

```diff
@@ -387,12 +394,13 @@ cdef inline void remove_var(float64_t val, float64_t *nobs, float64_t *mean_x,
             ssqdm_x[0] = 0
 
 
-def roll_var(const float64_t[:] values, ndarray[int64_t] start,
+def roll_var(float_complex_t[:] float_complex_values, ndarray[int64_t] start,
              ndarray[int64_t] end, int64_t minp, int ddof=1) -> np.ndarray:
     """
     Numerically stable implementation using Welford's method.
     """
     cdef:
+        float64_t[:] values = ensure_float64(float_complex_values)
```

Comment:
I don't think this is a suitable approach since it defeats the purpose of using a memory view in the input.